crypto/s2n_libcrypto.c

```diff
@@ -134,6 +138,19 @@ bool s2n_libcrypto_is_boringssl()
 #endif
 }
 
+S2N_RESULT s2n_libcrypto_init(void)
+{
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+    RESULT_ENSURE_NE(OSSL_PROVIDER_load(NULL, "default"), NULL);
+    /* needed to support RC4 algorithm
+     * https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html
+     */
+    RESULT_ENSURE_NE(OSSL_PROVIDER_load(NULL, "legacy"), NULL);
```

Comment:
These will return a pretty generic "S2N_ERR_SAFETY". Do you want a more specific error code?

Reply:
It should be ok since the operation is only doing one things (loading provider). We dont really have insight into why it might fail to load so I am not sure there is much to add other than "failed to load".

Reply:
But the customer will be calling s2n_init, not s2n_libcrypot_init. s2n_init does a lot of things :)